### PR TITLE
EIP1-1990: EIP1-2597: Implement happy path scenario for get certificate summary by application id

### PR DESCRIPTION
**Pull request by adacayi**
This PR is to implement the happy path scenario for getting certificate summary by application id.

### CODE REVIEW

File: src/main/kotlin/uk/gov/dluhc/printapi/database/mapper/PrintRequestStatusMapper.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    @ValueMapping(source = "PENDING_ASSIGNMENT_TO_BATCH", target = "PRINT_MINUS_PROCESSING")
+    @ValueMapping(source = "ASSIGNED_TO_BATCH", target = "PRINT_MINUS_PROCESSING")
+    @ValueMapping(source = "SENT_TO_PRINT_PROVIDER", target = "PRINT_MINUS_PROCESSING")
+    @ValueMapping(source = "RECEIVED_BY_PRINT_PROVIDER", target = "PRINT_MINUS_PROCESSING")
+    @ValueMapping(source = "VALIDATED_BY_PRINT_PROVIDER", target = "PRINT_MINUS_PROCESSING")
+    @ValueMapping(source = "IN_PRODUCTION", target = "PRINT_MINUS_PROCESSING")
+    @ValueMapping(source = "DISPATCHED", target = "DISPATCHED")
+    @ValueMapping(source = "NOT_DELIVERED", target = "NOT_MINUS_DELIVERED")
+    @ValueMapping(source = "PRINT_PROVIDER_VALIDATION_FAILED", target = "PRINT_MINUS_FAILED")
+    @ValueMapping(source = "PRINT_PROVIDER_PRODUCTION_FAILED", target = "PRINT_MINUS_FAILED")
```
Comment (Neil-Massey) [inline src/main/kotlin/uk/gov/dluhc/printapi/database/mapper/PrintRequestStatusMapper.kt:19]: https://technologyprogramme.atlassian.net/browse/EIP1-1990

`PRINT_PROVIDER_PRODUCTION_FAILED` maps to `PRINT_MINUS_PROCESSING`

File: src/test/kotlin/uk/gov/dluhc/printapi/rest/GetCertificateSummaryByApplicationIdIntegrationTest.kt [hunk 2]
```diff
@@ -70,19 +83,67 @@ internal class GetCertificateSummaryByApplicationIdIntegrationTest : Integration
 
     @Test
     fun `should return certificate summary given existing application and user with valid bearer token belonging to same ero and valid group`() {
```
Comment (Neil-Massey) [inline src/test/kotlin/uk/gov/dluhc/printapi/rest/GetCertificateSummaryByApplicationIdIntegrationTest.kt:85]: Is it worth adding a test for the `404` Not Found flow?

File: src/test/kotlin/uk/gov/dluhc/printapi/testsupport/testdata/entity/CertificateBuilder.kt [hunk 3]
```diff
@@ -96,7 +98,7 @@ fun buildPrintRequest(
     return printRequest
 }
 
-fun buildPrintStatus(
+fun buildPrintRequestStatus(
```
Comment (Neil-Massey) [inline src/test/kotlin/uk/gov/dluhc/printapi/testsupport/testdata/entity/CertificateBuilder.kt:101]: This rename touched a lot of files 😆

File: src/main/kotlin/uk/gov/dluhc/printapi/database/mapper/PrintRequestStatusMapper.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    @ValueMapping(source = "PRINT_PROVIDER_DISPATCH_FAILED", target = "PRINT_MINUS_FAILED")
```
[the remark this answers is not in the document]
Comment (adacayi) [reply src/main/kotlin/uk/gov/dluhc/printapi/database/mapper/PrintRequestStatusMapper.kt:20]: Thanks, @Neil-Massey ,
Seems like I misread this case and the below. I will fix it.

File: src/main/kotlin/uk/gov/dluhc/printapi/database/mapper/PrintRequestStatusMapper.kt (hunk 1 above)
Comment (adacayi) [reply src/main/kotlin/uk/gov/dluhc/printapi/database/mapper/PrintRequestStatusMapper.kt:19]: Thanks, @Neil-Massey ,
Seems like I misread this case and the above. I will fix it.

File: src/test/kotlin/uk/gov/dluhc/printapi/rest/GetCertificateSummaryByApplicationIdIntegrationTest.kt (hunk 2 above)
Comment (adacayi) [reply src/test/kotlin/uk/gov/dluhc/printapi/rest/GetCertificateSummaryByApplicationIdIntegrationTest.kt:85]: Definitely. I left it to be implemented as part of [EIP1-2593](https://technologyprogramme.atlassian.net/browse/EIP1-2593) subtask.

**Review by Neil-Massey — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/dluhc/printapi/rest/GetCertificateSummaryByApplicationIdIntegrationTest.kt (hunk 2 above)
Comment (adacayi) [reply src/test/kotlin/uk/gov/dluhc/printapi/rest/GetCertificateSummaryByApplicationIdIntegrationTest.kt:85]: I started working on it but saw it is dependent on the implementation of the happy path scenario, so paused and did the happy path first. I will have the PR for the not found scenario in a couple of minutes as well. 
Thanks for the approval.

**Pull request merged**